fluid_onnx/ops.py

```diff
@@ -665,10 +861,65 @@ def min_op():
 
 def matmul_op(operator, block):
     inputs, attrs, outputs = op_io_info(operator)
-    return make_node(
+    x_shape = block.vars[get_old_name(inputs['X'][0])].shape
+    y_shape = block.vars[get_old_name(inputs['Y'][0])].shape
+    name_matmul_x = inputs['X'][0]
+    name_matmul_y = inputs['Y'][0]
+    node_list = []
+    if attrs.get('transpose_X', False):
+        name_matmul_x, transposed_node = _matmul_transpose_op(name_matmul_x, block)
+        node_list.append(transposed_node)
+    if attrs.get('transpose_Y', False):
+        name_matmul_y, transposed_node = _matmul_transpose_op(name_matmul_y, block)
+        node_list.append(transposed_node)
+    name_matmul_out = outputs['Out'][0]
+    name_mul_out = name_matmul_out
+    need_unsqueeze = len(x_shape) == 1 and len(y_shape) == 1
+    need_to_scale = attrs.get('alpha', None) != 1.0
+    unsqueeze_inputs = []
+    if need_to_scale or need_unsqueeze:
+        name_matmul_out = _tmp_name(name_matmul_out, 'matmul')
+    if need_to_scale and need_unsqueeze:
+        name_mul_out = name_matmul_out + '_scaled'
+        unsqueeze_inputs = [name_mul_out]
+    elif not need_to_scale and need_unsqueeze:
+        name_mul_out = _tmp_name(outputs['Out'][0], 'scaled')
+        unsqueeze_inputs = [name_matmul_out]
+    node_list.append(make_node(
         'MatMul',
-        inputs=inputs['X'] + inputs['Y'],
-        outputs=outputs['Out'])
+        inputs=[name_matmul_x, name_matmul_y],
+        outputs=[name_matmul_out]))
+    if need_to_scale:
+        scale_const_node = global_const(np.array(attrs['alpha'], dtype=np.float32), block)
+        node_list.extend([
+            scale_const_node.node,
+            make_node(
+                'Mul',
+                inputs=[name_matmul_out, scale_const_node.name],
+                outputs=[name_mul_out])
+        ])
+    if need_unsqueeze:
+        rank1_shape_const = global_const(np.array([1], dtype=np.int64), block, node_name='rank1_shape')
+        node_list.append(rank1_shape_const.node)
+        node_list.append(onnx.helper.make_node(
+            'Reshape',
+            inputs=unsqueeze_inputs + [rank1_shape_const.name],
+            outputs=outputs['Out']))
```

Comment:
ONNX `Unsqueeze` on scalar would change it to a rank 2 tensor like `[[value]]`, so use `Reshape` instead